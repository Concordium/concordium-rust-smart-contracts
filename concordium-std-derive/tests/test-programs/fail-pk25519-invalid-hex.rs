//! Ensure that the macros generate compilable code.

use concordium_contracts_common::*;
use concordium_std_derive::*;

const PK_25519: PublicKeyEd25519 = public_key_ed25519!("0l2a7e286063ae5dfcebce40636c0546d367d8c65cd4cb69aae1af77a4d61412");

fn f() {
    println!("{:?}", PK_25519.to_string());
}

fn main() {}
