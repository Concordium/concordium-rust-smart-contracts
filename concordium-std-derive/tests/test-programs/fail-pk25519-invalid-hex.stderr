error: Invalid Ed25519 public key: Parsing failed
 --> tests/test-programs/fail-pk25519-invalid-hex.rs:6:56
  |
6 | const PK_25519: PublicKeyEd25519 = public_key_ed25519!("0l2a7e286063ae5dfcebce40636c0546d367d8c65cd4cb69aae1af77a4d61412");
  |                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
